asserts/repair.go

```diff
@@ -0,0 +1,81 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package asserts
+
+import (
+	"fmt"
+)
+
+// Repair holds an repair assertion which allows running repair
+// code to fixup broken systems. It can be limited by series and models.
+type Repair struct {
+	assertionBase
+
+	series []string
+	models []string
+}
+
+// RepairID returns the "id" of the repair. It should be a short string
+// that follows a convention like "REPAIR-123". Similar to a CVE there
+// should be a public place to look up details about the repair-id
+// (e.g. the snapcraft forum).
+func (em *Repair) RepairID() string {
```

Comment:
I see two approaches either 1) we have clever a service that gives us back all relevant repairs,  

2) or we say that devices are sort of subscribed to sequences of repairs,  all would be subscribed to (canonical, repair-#) at least,  than we would fetch the next (increased seq num for each of them)

a cons of 1 is that we need to think exactly how to not return a huge list, what kind of information to send to let the other side do filtering for us, the more complex information we need to send the more it's fragile because we might not be able to find the information exactly because we are broken

a cons of 2 is that it seems to need some integration into the process of image building/configuration, we need to list streams at that point somehow, because ideally we want to both give a stream and a starting point (reapairs older than that are known to be not relevant), we can specify that when we build new snapd/core for the default streams but not for the per device/brand ones